File: Python/Valid Parentheses/ValidParentheses.py
class Solution:
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        oppo = (')', ']', '}')
        if(len(s) % 2 != 0):
            return False
        for c in s:
            pos = '([{'.find(c)
            if(pos >= 0):
                stack.append(oppo[pos])
            elif(stack==[] or stack.pop() != c):
                return False
        return stack == []

File: C#/Valid Parentheses/ValidParentheses.cs
public class Solution {
    public bool IsValid (string s) {
        if (s.Length % 2 != 0)
            return false;
        char c;
        Stack<char> stack = new Stack<char>();
        for (int i = 0; i < s.Length; i++) {
            c = s[i];
            if (c == '(')
                stack.Push (')');
            else if (c == '{')
                stack.Push ('}');
            else if (c == '[')
                stack.Push (']');
            else if (stack.Count == 0 || stack.Pop() != c)
                return false;
        }
        return stack.Count == 0;
    }
}